src/libraries/Microsoft.PowerFx.Core/Entities/External/IExternalNamedFormulaRule.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using Microsoft.PowerFx.Core.App.Controls;
+using Microsoft.PowerFx.Core.Lexer.Tokens;
+using Microsoft.PowerFx.Core.Syntax;
+using Microsoft.PowerFx.Core.Types;
+
+namespace Microsoft.PowerFx.Core.App
+{
+    /// <summary>
+    /// Wraps a named formula's Formula, Rule, Token, and ResultType.
+    /// </summary>
+    internal interface IExternalNamedFormulaRule
```

Comment:
>IExternalNamedFormulaRule [](http://example.com/codeflow?start=23&length=25)

Can we use the IExternalRule interface for named formulas, instead of having a separate interface?

Reply:
we chatted offline, and we'll avoid this new interface and instead have a flag on the NameLookupInfo.